### PR TITLE
Testing a theory

### PR DESCRIPTION
**Pull request by stkem**
Android can no longer connect to the web socket and this is the only
relevant recent server side change I can find.

@jaredpetker @2is10

### CODE REVIEW

**Comment by stkem** [comment]
Theory seems to have been correct. On Monday (or tomorrow if you guys want to do it without me) we need to figure out 
a) Why this was failing (i.e. is the version parser on the backend bad or is the app sending something bad for the version)
b) Why it was failing silently.

**Comment by 2is10** [comment]
Thanks Stephen. Will investigate.

**Comment by 2is10** [comment]
One problem I see is that UserAgent declares a `KifiAndroidAppTypeName` constant but doesn’t ever actually use it. Maybe as a result the server thinks it’s a browser extension connection instead.

@jaredpetker We also need to confirm that the version string the Android app uses conforms what the server is expecting. See `KifiAndroidVersion` in KifiVersion.scala.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala [hunk 1]
```diff
@@ -168,10 +168,17 @@ trait AuthenticatedWebSocketsController extends ElizaServiceController {
         case Some(streamSessionFuture) => streamSessionFuture.map { streamSession =>
```
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: This is likely why it was failing silently, by the way.

Comment (2is10) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: What’s the recommended fix? Wrap in `new SafeFuture`?

Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: Oh, I missed the original comment here. I think it's either that or attach an error handler / recover to the future after the map. I would vote for the latter in this case, i.e. recover, airbrake and/or log the exception and then continue with some placeholder/default version.

**Comment by 2is10** [comment]
I spoke with @jaredpetker about Android app versioning today. He said he’s been using just two numbers (major.minor) and the server was expecting three or four (fixed in #10262). The Android app was also using a "ka" prefix on the version parameter value, like `?version=ka1.5` but JP is removing the "ka". After we confirm the fixes are working, I guess we can remove the try/catch.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: Hmm... I didn’t author that part and I think you’d do a better job, and quicker. You do it and I’ll review?

Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: Sure. PR coming soon.

Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/AuthenticatedWebSocketsController.scala:168]: Actually it turns out that future is returning a rather complex result (it's not just the version) where we can't easily fall back on some default.
I'll make sure that it's no longer silent, but a bad version will become a fatal error on connect again (as it was before). I think that might actually be a good thing for a little bit of protection against tampered with clients.

**Pull request merged**